### PR TITLE
Refactor PDF report headings

### PR DESCRIPTION
**Pull request by bangbay-bluetiger**
### Description

- Update headings in report PDF view for WP and SAR
- Calculate child heading levels from parent
- Add `headingLevel` to components and define `HeadingLevel` type
- Update and add unit tests

### Related ticket(s)

CMDCT-3678

---
#### Local
1. Start MFP
2. Create a WP and SAR (use `yarn db:seed`)

#### Staging
1. Login as stateuser at [https://d7jgww7pwtt4z.cloudfront.net/](https://d7jgww7pwtt4z.cloudfront.net/)
2. WP and SAR have already been created

#### Next steps
3. Navigate to `Review PDF` button for both WP and SAR
4. Verify tags match hierarchy defined in [doc](https://docs.google.com/document/d/1gPWBOTqFyzsZZdyD6h3absIxZmkHmfLWEocwdKG_N4A/edit#heading=h.lwh6luwjoym6) 
6. On Chrome, you can use a HTML visualization extension to highlight heading tags. Example with HTML Visual Validation:
Screenshot 2024-06-20 at 11 50 20 AM

### Notes

---
### Pre-review checklist

- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [x] I have updated relevant documentation, if necessary
- [x] I have performed a self-review of my code
- [x] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [x] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

### CODE REVIEW

**Review by gmrabian — APPROVED** [review]
🚀

File: services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx [hunk 1]
```diff
@@ -85,12 +90,14 @@ export function getEntityTableComponents(
   return (
     <Box key={uuid()}>
       <Box>
-        <Heading as="h4">
-          <Box sx={sx.stepName}>{title}</Box>
+        <Box>
```
Comment (jessabean) [inline services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx:93]: Do we need the extra div here?

**Review by jessabean — APPROVED** [review]
Looks good! I have a question about why we need extra `<Box>` wherever it was added, but everything is working as expected.

File: services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx (hunk 1 above)
Comment (bangbay-bluetiger) [reply services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx:93]: Good catch. No, we don't need it.

**Review by gmrabian — APPROVED** [review]

**Review by jessabean — APPROVED** [review]

**Pull request merged**